### PR TITLE
Thellimist/pi 785

### PR DESCRIPTION
**Pull request by thellimist**

### CODE REVIEW

File: src/API/AbstractAPIClient.php [hunk 1]
```diff
@@ -51,9 +53,18 @@ public function callAPI(Request $request)
...
-            $apiRequest = $client->createRequest($request->getMethod(), $request->getUrl(), $requestOptions);
+            $apiRequest = $client->createRequest($request->getMethod(), $request->getUrl(), $request->getHeaders(), $request->getBody(), $request->getParameters());
 
-            $response = $apiRequest->send($apiRequest)->json();
+            $method = $request->getMethod();
+
+            // Since Guzzle automatically overwrites a new header when the request
+            // is POST / PATCH / DELETE, we need to overwrite the Content-Type header
+            // with setBody() function.
+            if ($method === 'PATCH' || $method === 'DELETE' || $method === 'POST') {
```
Comment (jwineman) [inline src/API/AbstractAPIClient.php:63]: We don't need `PUT` here?

Comment (thellimist) [reply src/API/AbstractAPIClient.php:63]: We do. Nice catch

Comment (jwineman) [reply src/API/AbstractAPIClient.php:63]: Probably easier to do

```
if($method !== 'GET') 
```

Comment (thellimist) [reply src/API/AbstractAPIClient.php:63]: I'm not certain. Except for `POST` nothing is documented. I had to discover `PATCH` and `DELETE` myself. To be honest we can just override the Content-Type to `'application/json'` in all cases.

Comment (thellimist) [reply src/API/AbstractAPIClient.php:63]: https://github.com/cloudflare/cloudflare-plugin-backend/pull/26

**Pull request merged**